README.md

```diff
@@ -1,35 +1,21 @@
+Pulling image is one of the time-consuming steps in the container lifecycle. Research shows that time to take for pull operation accounts for 76% of container startup time[[FAST '16]](https://www.usenix.org/node/194431). *Stargz Snapshotter* is an implementation of snapshotter which aims to solve this problem leveraging [stargz image format by CRFS](https://github.com/google/crfs). The following histogram is the benchmarking result for startup time of several containers measured on Github Actions, using Docker Hub as a registry.
 
-Related discussion of the snapshotter in containerd community:
-- [Support remote snapshotter to speed up image pulling#3731@containerd](https://github.com/containerd/containerd/issues/3731)
-- [Support `Prepare` for existing snapshots in Snapshotter interface#2968@containerd](https://github.com/containerd/containerd/issues/2968)
-- [remote filesystem snapshotter#2943@containerd](https://github.com/containerd/containerd/issues/2943)
+<img src="docs/images/benchmarking-result-288c338.png" width="600" alt="The benchmarking result on 288c338">
 
-By using this snapshotter, images(even if they are huge) can be pulled in lightning speed because this skips pulling layers but fetches the contents on demand at runtime.
-```
-# time ctr-remote images rpull --plain-http registry2:5000/fedora:30 > /dev/null 
-real	0m0.447s
-user	0m0.081s
-sys	0m0.019s
-# time ctr-remote images rpull --plain-http registry2:5000/python:3.7 > /dev/null 
-real	0m1.041s
-user	0m0.073s
-sys	0m0.028s
-# time ctr-remote images rpull --plain-http registry2:5000/jenkins:2.60.3 > /dev/null 
-real	0m1.231s
-user	0m0.112s
-sys	0m0.008s
-```
+`legacy` shows the startup performance when we use containerd's default snapshotter (`overlayfs`) with images copied from `docker.io/library` without optimization. For this configuration, containerd pulls entire image contents and `pull` operation takes accordingly. When we use stargz snapshotter with `stargz` images we are seeing performance improvement on the `pull` operation because containerd can start the container before the entire image contents locally available but fetches each file on-demand. When we use the heavily optimized version of images(`estargz`) the performance of `run` operation (= running ["hello world"-like programs](./script/benchmark/hello-bench/src/hello.py) of each image) is much faster than `stargz`. This is because stargz snapshotter prefetches and caches some files which will be most likely accessed during container execution. Stargz snapshotter waits for the first container creation until the prefetch completes so `create` sometimes takes longer than other types of image. But this wait only occurs just after the pull completion until the prefetch completion and it's much shorter than waiting for downloading all files of all layers.
```

Comment:
Is `estargz` "much" faster than `stargz`?

Reply:
I agree with `much faster` is misleading. I think `we can mitigate the performance drawback observed in stargz images` is the more precise expression with the following context:

> ... When we use stargz snapshotter with `stargz` images we are seeing performance improvement on the `pull` operation because containerd can start the container before the entire image contents locally available and fetches each file on-demand. But at the same time, we see the performance drawback for `run` operation because each access to files takes extra time for fetching them from the registry. When we use the further optimized version of images(`estargz`) we can mitigate the performance drawback observed in `stargz` images. ...

Reply:
looks good